docs/source/basic_concepts.rst

```diff
@@ -2,60 +2,88 @@
  Basic Concepts
```

Comment:
It is better to use "refine" rather than "modify" in the commit message.
here is my idea of the commit message:
```
docs: refine the contents of "Basic Concepts" and "Features" chapters
```